wgengine/wgengine.go

```diff
@@ -65,6 +66,9 @@ type Engine interface {
 	// SetFilter updates the packet filter.
 	SetFilter(*filter.Filter)
 
+	// SetDNSMap updates the DNS map.
```

Comment:
say something about ownership? that neither party should mutate it after this call?

perhaps you should just make the type immutable to make that more clear, and make a constructor that returns the immutable type? (unexporting the map)

Reply:
The intent is to eventually add more fields to `tsdns.Map` as needed (e.g. a custom parent domain instead of `.ipn.dev`). In that case, the constructor may become a bit unwieldy. Should it perhaps be passed by value instead?